lib/ddtrace/writer.rb

```diff
@@ -79,29 +78,20 @@ def send_spans(traces, transport)
       # Inject hostname if configured to do so
       inject_hostname!(traces) if Datadog.configuration.report_hostname
 
-      if transport.is_a?(Datadog::HTTPTransport)
-        # For older Datadog::HTTPTransport...
-        code = transport.send(:traces, traces)
-        (!transport.server_error?(code)).tap do |status|
-          @traces_flushed += traces.length if status
-        end
-      else
-        # For newer Datadog::Transports...
-        # Send traces an get a response.
-        response = transport.send_traces(traces)
-
-        unless response.internal_error?
-          @traces_flushed += traces.length unless response.server_error?
-
-          # Update priority sampler
-          unless priority_sampler.nil? || response.service_rates.nil?
-            priority_sampler.update(response.service_rates)
-          end
-        end
+      # Send traces an get a response.
+      response = transport.send_traces(traces)
+
+      unless response.internal_error?
+        @traces_flushed += traces.length unless response.server_error?
 
-        # Return if server error occurred.
-        !response.server_error?
+        # Update priority sampler
+        unless priority_sampler.nil? || response.service_rates.nil?
+          priority_sampler.update(response.service_rates)
+        end
       end
+
+      # Return if server error occurred.
```

Comment:
This feels a little misleading. It looks like we return whether or not a server error was absent. IE if there was a server error we return false and if there was not we will return true.

Reply:
Yeah maybe the comment isn't as accurate, but the return value looks correct.